### ISSUE TITLE
Database table configuration either undocumented or doesn't work

### ISSUE THREAD

**Issue by brandondrew** [hschne/rails-mini-profiler#27]
### Issue Summary

I am trying to use different names for the database tables.  I don't know whether the existing documentation is incomplete or whether the table name configurations are failing.  I tried the most obvious configuration

```
config.enabled                 = true
config.badge_enabled           = true
config.badge_position          = 'top-left'
config.skip_paths              = []
config.profiled_requests_table = :profiler_requests       # these get errors
config.flamegraphs_table       = :profiler_flamegraphs    # when starting
config.traces_table            = :profiler_traces         # Rails
config.flamegraph_enabled      = true
config.flamegraph_sample_rate  = 0.0125
```

There was a hint that these 3 configurations were meant to be nested together under 'storage', but there were no examples.   I tried

```
  config.enabled                 = true
  config.badge_enabled           = true
  config.badge_position          = 'top-left'
  config.skip_paths              = []
  config.storage do |conf|
    conf.profiled_requests_table = :profiler_requests
    conf.flamegraphs_table       = :profiler_flamegraphs
    conf.traces_table            = :profiler_traces
  end
  config.flamegraph_enabled      = true
  config.flamegraph_sample_rate  = 0.0125
```

This was just a guess, and it allowed me to start up Rails, but in turn I got 

```
PG::UndefinedTable: ERROR: relation "rmp_profiled_requests" does not exist LINE 8: WHERE a.attrelid = '"rmp_profiled_requests"'::regclass ^
```

### Steps to Reproduce

1. Use the configs above
2. Observe errors

### Technical details

* Rails version: `rails --version`
* Ruby version: `ruby 2.7.1p83 (2020-03-31 revision a0c7c23c9c) [x86_64-darwin19]`

**Comment by brandondrew** [comment]
Just noticed another clue in the config file, and tried this:

```
  config.storage = RailsMiniProfiler::Storage do |conf|
    conf.profiled_requests_table = :profiler_requests
    conf.flamegraphs_table       = :profiler_flamegraphs
    conf.traces_table            = :profiler_traces
  end
```

It also failed.

**Comment by brandondrew** [comment]
I noticed by poking around in Pry that  there's a `RailsMiniProfiler.storage_configuration` method, so I tried

```
  config.storage = RailsMiniProfiler::Storage
  config.storage_configuration.profiled_requests_table = :profiler_requests
  config.storage_configuration.flamegraphs_table       = :profiler_flamegraphs
  config.storage_configuration.traces_table            = :profiler_traces
```

but that got a response of 

> \`defined method \`storage_configuration' for #<RailsMiniProfiler::Configuration:0x00007fd342d70e48> (NoMethodError)\`

**Comment by brandondrew** [comment]
I tried putting this outside of the `|config|` block:

```
RailsMiniProfiler.storage_configuration.profiled_requests_table = :profiler_requests
RailsMiniProfiler.storage_configuration.flamegraphs_table       = :profiler_flamegraphs
RailsMiniProfiler.storage_configuration.traces_table            = :profiler_traces
```

and this gave me

> ActiveModel::MissingAttributeError in UcbRailsUser::UsersController#edit
can't write unknown attribute `rmp_profiled_request_id

**Comment by hschne** [comment]
Thanks for the detailed write-up, it's super helpful! :raised_hands: 

You are right, the documentation here is definitely lacking. I looked into this briefly, there seem to be some issues with storage configuration in general at the moment. I think you may be able to get this to work with some small modifications for now: 

Try updating your configuration to this: 

```ruby
RailsMiniProfiler.configure do |config|
  config.storage.profiled_requests_table = 'my_new_table_name'
end
```

Now, there is a bug that results in things breaking here. But it's an easy fix. Using `bundle open rails_mini_profiler`, you can open the file `lib/rails_mini_profiler.rb` and update `storage_configuration` like so:
```ruby
def storage_configuration
  # remove the last 'configuration' from 'configuration.storage.configuration'
  configuration.storage
end
```
Regarding the original error you got: Did you run the migrations, and is a table `rmp_profiled_requests` available? :thinking: 

Feel free to open a PR with documentation improvements (if you want). I'll try and get a fix out soon :slightly_smiling_face:

**Comment by hschne** [comment]
@brandondrew This issue should be resolved in 0.2.0. Documentation has been improved, hopefully it is more clear how storage can be configured. If you face any other issues please do not hesitate to let me know. 

Thanks again for contributing to this project! :bow:

**Comment by olleolleolle** [comment]
I ran into this issue, and upgrading from 0.1-something, I needed to pass an instance to the configuration for storage:



```diff
# Use RailsMiniProfiler::Storage::ActiveRecord to store profiles in your Database
- config.storage = RailsMiniProfiler::Storage
+ config.storage = RailsMiniProfiler::Storage.new
```
